### ISSUE TITLE
[NextJS] Migrate page routes to app routes

### ISSUE THREAD

**Issue by usame-algan** [safe-global/safe-wallet-web#2541]
## What is the feature about

Next 13 introduced app routes which are meant to replace page routes and there are performance optimizations coming out for app routes: https://nextjs.org/blog/next-13-5?trk=feed_main-feed-card-text#improving-startup-and-fast-refresh-time

## The list of requirements

- Follow the migration guide: https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration

**Comment by usame-algan** [comment]
One benefit I found is that we can fetch data globally via the `RootLayout` instead of having to add `getStaticProps` to every page

**Comment by usame-algan** [comment]
Is this issue a pre-requisite for using turbopack @compojoom?

**Comment by compojoom** [comment]
I don't think so. In my tests of turbopack out app was failing to compile for other reasons and not page routes.
